Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs

```diff
@@ -47,24 +47,26 @@ public override void OnReload(GunStats stats)
     public override void OnFire(GunStats stats)
     {
         animator.SetTrigger("Fire");
-        for (int i = stats.Ammo; i < magazineSize; i++)
+        for (int i = stats.Ammo - 1; i < magazineSize; i++)
         {
             ammunition[i].SetActive(false);
         }
-        if (stats.Ammo == 0)
+        if (stats.Ammo == 1)
         {
-            ToggleBullet();
+            bullet.SetActive(false);
```

Comment:
This `if` statement doesn't really do anything now, it just sets the bullet inactive right before the animation calls `EnableBullet`

Reply:
I had to add a state due to timing restrictions, but have updated it now 👍